src/main/java/hudson/scm/IntegrityCheckpointAction.java

```diff
@@ -74,16 +79,17 @@ public IntegrityCheckpointAction(String serverConfig, String checkpointLabel)
    * @param env Environment containing the name/value pairs for substitution
    * @param expression Groovy expression string
    * @return Resolved string
+   * @throws Exception 
    */
-  public static String evalGroovyExpression(Map<String, String> env, String expression)
+  public static String evalGroovyExpression(Map<String, String> env, String expression) throws Exception
   {
     Binding binding = new Binding();
-    binding.setVariable("env", env);
-    binding.setVariable("sys", System.getProperties());
-    CompilerConfiguration config = new CompilerConfiguration();
-    // config.setDebug(true);
-    GroovyShell shell = new GroovyShell(binding, config);
-    Object result = shell.evaluate("return \"" + expression + "\"");
+	binding.setVariable("env", env);
+	// binding.setVariable("sys", System.getProperties());
+	ClassLoader loader = GroovySandbox
+			.createSecureClassLoader(Jenkins.getActiveInstance().getPluginManager().uberClassLoader);

```

Comment:
`evaluate` already wraps the supplied loader, so you should not call `createSecureClassLoader` yourself in this case.

Reply:
Dear @jglick,
Noted. Will be fixed in the next release